packages/bitcore-p2p/lib/pool.js

```diff
@@ -54,7 +54,7 @@ function Pool(options) {
   this.dnsSeed = options.dnsSeed !== false;
   this.maxSize = options.maxSize || Pool.MaxConnectedPeers;
   this.messages = options.messages;
-  this.network = Networks.get(options.network) || Networks.defaultNetwork;
+  this.network = (this.messages && this.messages.network) || Networks.get(options.network) || Networks.defaultNetwork;
```

Comment:
you may want to swap the ordering around to
```
this.network = Networks.get(options.network)|| (this.messages && this.messages.network) || Networks.defaultNetwork;
```
so that in current usage everywhere it will follow what it was using, and anywhere we want it to follow the messages, we just omit network from the pool construction.

Reply:
We're gonna close this in favor of this: https://github.com/bitpay/bitcore/pull/3091